api/base/filters.py

```diff
@@ -171,7 +188,11 @@ def param_queryset(self, query_params, default_queryset):
         if fields_dict:
             for field_name, value in fields_dict.items():
                 if self.is_filterable_field(key=field_name):
-                    queryset = queryset.intersection(set(self.get_filtered_queryset(field_name, value, default_queryset)))
+                    filtered_queryset = set(self.get_filtered_queryset(field_name, value, default_queryset))
+                    queryset = [
+                        result for result in queryset
+                        if result in filtered_queryset
+                    ]
```

Comment:
Agreed this code is confusing. And I may have accidentally inverted the meaning. 

Trying to think this through:

1. `queryset` needs to be the intersection of each of the `filtered_queryset`s, and starts out as the unfiltered `default_queryset`. 
2. Each `filtered_queryset` is a subset of `default_queryset`. So at each iteration of the loop we want to keep all of the members of `queryset` that are also members of `filtered_queryset`
```python
[result for result in queryset if result in filtered_queryset]
```
3. `queryset` is iteratively reduced by the constraint above

Does this help at all?